flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/reader/TaskEventSender.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.api.reader;
+
+import org.apache.flink.runtime.event.TaskEvent;
+
+import java.io.IOException;
+
+/**
+ * The basic API for every reader.
+ */
+public interface TaskEventSender {
+
+	void sendTaskEvent(TaskEvent event) throws IOException;
```

Comment:
Currently `sendTaskEvent` is only used for iteration scenarios, so I think it may also make sense if covering in `IterationReader` similar with `registerTaskEventListener`. Only when we consider to extend other `TaskEvent` in future to make it a separate interface for common cases.

Reply:
Yeah, maybe we just simply change the name of ```ReaderBase``` to ```IterationReader```